File: QE-Java Selenium/Activity10.java
package demos;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.*;
import java.time.Duration;

public class Activity10 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://training-support.net/webelements/drag-drop");

        System.out.println(driver.getTitle());

        WebElement ball = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("ball")));
        WebElement dz1  = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("dropzone1")));
        WebElement dz2  = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("dropzone2")));

        Actions a = new Actions(driver);

        a.clickAndHold(ball).moveToElement(dz1).release().perform();
        System.out.println("DZ1: " + dz1.getText());

        a.clickAndHold(ball).moveToElement(dz2).release().perform();
        System.out.println("DZ2: " + dz2.getText());

        driver.quit();
    }
}

File: QE-Python Selenium/Activity10.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)
driver.get("https://training-support.net/webelements/drag-drop")

print(driver.title)

ball = wait.until(EC.visibility_of_element_located((By.ID, "ball")))
dz1  = wait.until(EC.visibility_of_element_located((By.ID, "dropzone1")))
dz2  = wait.until(EC.visibility_of_element_located((By.ID, "dropzone2")))

a = ActionChains(driver)

for dz, name in [(dz1, "Dropzone 1"), (dz2, "Dropzone 2")]:
    a.click_and_hold(ball).move_to_element(dz).release().perform()
    assert "Dropped" in dz.text
    print(f"Ball entered {name}")

driver.quit()
